utils/generate-playbook

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python3
+import argparse
+from argparse import RawTextHelpFormatter
+import re
+import sys
+from os.path import abspath, dirname, join
+from ruamel import yaml
+from operator import itemgetter
+from collections import defaultdict
+
+# Import common_koji from the local "module_utils" directory.
+working_directory = dirname(abspath((__file__)))
+module_utils_path = join(dirname(working_directory), 'module_utils')
+if module_utils_path not in sys.path:
+    sys.path.insert(0, module_utils_path)
+import common_koji  # NOQA: E402
+
+
+DESCRIPTION = """
+This tool queries Koji for all the tags and their settings.
+It then writes an Ansible playbook to STDOUT.
+
+Use this tool to quickly get started with koji-ansible, or to transfer
+a configuration from one Koji instance to another.
+"""
+
+
+def interrogate(session, tags):
+    """
+    Interrogate Koji for all information about these tags.
+
+    Quickly get all the information we need from Koji with as few XML-RPC
+    calls as possible.
+    """
+    # Note: This client multiCall method bombs out after about 35 minutes for
+    # large Koji task lists (tested with 11,170 tags). If our tags list is
+    # very large, we need to chunk it up here before we send a
+    # multicall-of-death. This will take some tuning as we add more features
+    # (additional calls) to this code as well.
+    session.multicall = True
+    for tag in tags.copy():
+        tag_name = tag['name']
+        session.getInheritanceData(tag_name)
+        session.listPackages(tag_name)
+        session.getTagExternalRepos(tag_name)
+    multicall_results = session.multiCall(strict=True)
+    session.multicall = False
```

Comment:
You don't need to explicitly set session.multicall to False - multiCall() will do it by itself.

Reply:
Thank you. I switched to the new multi-call interface in Koji 1.18 and this code is better now.